docs/source/examples/classification_and_regression.rst

```diff
@@ -0,0 +1,303 @@
+
+.. code-block:: python
+
+    class ClassifierLocalizer(nn.Module):
+        def __init__(self, model_name, num_classes=2):
+            super(ClassifierLocalizer, self).__init__()
+            self.num_classes = num_classes
+            
+            # create cnn model
+            model = getattr(models, model_name)(pretrained=True)
+            
+            # remove fc layers and add a new fc layer
+            num_features = model.fc.in_features
+            model.fc = nn.Linear(num_features, 6) # classifier + localizer
+            self.model = model
+        
+        def forward(self, x):
+            x = self.model(x)                    # extract features from CNN
+            scores = x[:, :self.num_classes]     # class scores
+            coords = x[:, self.num_classes:]     # coordinates
+            return [scores, torch.sigmoid(coords)]   # sigmoid output is in the range of [0, 1]
+
+Regarding the complexity of the problem, the number of the samples in the training dataset, and the similarity of the training dataset to the ImageNet dataset, we may decide to freeze some of the layers. In our current example, based on the mentioned factors, we freeze all layers but the last fully connected layer.
+
+.. code-block:: python
+
+    network = ClassifierLocalizer(model_name='resnet18')
+
+    def freeze_weights(network):
+        for name, param in network.named_parameters():
+            if not name.startswith('fc.'):
+                param.requires_grad = False
+
+    freeze_weights(network)
+    print(network)
+
+Loss function
+=============
+
+As we discussed before, we have two different tasks in this example. These tasks need different loss functions; Cross-Entropy loss for the classification and Mean Square Error loss for the regression. Below, we define a new loss function class that sums both losses to considers them simultaneously. However, as the regression is relatively a harder task, we apply a higher weight to MSEloss.
+
+.. code-block:: python
+
+    class ClassificationRegressionLoss(nn.Module):
+        def __init__(self):
+            super(ClassificationRegressionLoss, self).__init__()
+            self.ce_loss = nn.CrossEntropyLoss() # size_average=False
+            self.mse_loss = nn.MSELoss()
+            
+        def forward(self, y_pred, y_true):
+            loss_cls = self.ce_loss(y_pred[0], Variable(y_true[0][:,20])) # Cross Entropy Error (for classification)
+            loss_reg1 = self.mse_loss(y_pred[1][:,0], y_true[1][:,0]/h) # Mean Squared Error for X_L
+            loss_reg2 = self.mse_loss(y_pred[1][:,1], y_true[1][:,1]/w) # Mean Squared Error for Y_L
+            loss_reg3 = self.mse_loss(y_pred[1][:,2], y_true[1][:,2]/h) # Mean Squared Error for X_R
+            loss_reg4 = self.mse_loss(y_pred[1][:,3], y_true[1][:,3]/w) # Mean Squared Error for Y_R
+            total_loss = loss_cls + W * (loss_reg1 + loss_reg2 + loss_reg3 + loss_reg4)
```

Comment:
`W` should be parameter of the `__init__` method of the class, not a global as currently.